qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java

```diff
@@ -214,18 +207,58 @@ public void shareFileTest() {
         Spoon.screenshot(mActivity, "after");
     }
 
+    private void goToFiles() {
+        openDrawer(R.id.drawer_layout);
+        onView(withText(R.string.filebrowser)).check(matches(isDisplayed())).perform(click());
+    }
+
     @Test
     public void sendFileTest() {
         Spoon.screenshot(mActivity, "startup");
         onView(withText(exampleFiles.get(0).getName())).perform(longClick());
 
         onView(withText(R.string.Send)).perform(click());
 
-        UITestHelper.sleep(1000);
-
         //Check Chooser
         intended(allOf(hasAction(Intent.ACTION_CHOOSER), hasExtra(Intent.EXTRA_TITLE, mActivity.getString(R.string.share_via))));
     }
 
+    private void createFolder() {
+        onView(withId(R.id.fab)).check(matches(isDisplayed())).perform(click());
+        onView(withText(R.string.create_folder)).check(matches(isDisplayed())).perform(click());
+
+        onView(allOf(withClassName(endsWith("EditTextFont")))).perform(typeText(CREATE_FOLDER_TEST_NAME));
+        onView(withText(R.string.ok)).perform(click());
+    }
+
+    @Test
+    public void testCreateFolder() {
+        goToFiles();
```

Comment:
Testing the drawer is not useful. You can add an Intent-extra to the launch Intent of the MainActivity which brings you directly to the files fragment. (It doesn't exist yet ;) )